src/core/mst-operations.ts

```diff
@@ -891,8 +891,8 @@ export function getMembers(target: IAnyStateTreeNode): IModelReflectionData {
       else reflected.volatile.push(key)
       return
     }
-    if (descriptor.value._isMSTAction === true) reflected.actions.push(key)
     if (descriptor.value._isFlowAction === true) reflected.flowActions.push(key)
+    if (descriptor.value._isMSTAction === true) reflected.actions.push(key)
```

Comment:
**praise**: this is such a subtle bug. I had to read it a few times over. This has gotten me in other code bases before.

The problem is that we originally constructed a conditional statement like "If this is an MST action, push it into the actions block. Otherwise, if it's observable, we push it into the volatiles, and if that's not true, then this is a view and should be pushed there. The clauses were:

```ts
if (descriptor.value._isMSTAction == true) {
 . . .
 } else if (isObservableProp(target, key)) {
  . . . 
 } else {
  . . . 
 }
```

But when we added the [`_isFlowAction` check](https://github.com/mobxjs/mobx-state-tree/commit/62e7e8bafe8bf07f4694e429f734bb12323fd8ef#diff-3090c65cf4a55928afed579b85369bf9c433e2c9bd615adce4127686a944b56aR874), we made the original `._isMSTAction` no longer part of the `if / else`. So now the clauses are:

```ts
if (descriptor.value._isMSTAction == true) { } // Stands alone

// Separate if/else logic, and an _isMSTAction will also try to evaluate this one.
if (descriptor.value._isFlowAction == true) {
 . . .
 } else if (isObservableProps(target, key)) {
  . . . 
 } else {
  . . . 
 }
 ```
 
 So anything where `_isMSTAction === true` will get pushed into the `actions` array, and then try to evaluate itself in the second block of logic. It falls through to the final `else` clause and gets added to the `views` array.
 
@evinosheaforward thank you for the eyes on this one. I have a feeling this would have taken me far too long to spot.

**suggestion (non-blocking):** @evinosheaforward - On that note, will you please change the formatting of this whole conditional clause to use brackets, rather than the if/else shorthand? I think that will help prevent this issue in the future. As I mentioned in my first comment, I've hit this exact bug in other code bases before, and I find that it usually comes from too much shorthand, which can make it hard to spot the logic flaw.

Reply:
Yeah, very easy bug to write.
Since I was coming at it from the perspective of _knowing the bug_ it stood out.

I updated the syntax/formatting for clarify, good idea.